### ISSUE TITLE
Parsing Error when using Nullish Assignment (??=)

### ISSUE THREAD

**Issue by stephen-j-oleary** [browserify/browserify#2029]
I'm getting the following error when attempting to bundle my project.
`Error: Parsing file /path/to/file.js: Unexpected token (13:22)` 
I've traced the error to a Nullish Assignment (??=) [.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment) Changing that line to a regular assignment allows the project to build without any errors. 

Is there any way to add support for this type of assignment operator or should I just avoid using it?

Thanks

**Comment by ljharb** [comment]
What version of node are you using? If it doesn't support it natively, it can't work without transpilation.

Either way, you probably want to be transpiling (with babel) so that you can target older envs.

**Comment by stephen-j-oleary** [comment]
Thanks for the response. I'm using node v16.13.2 for this project. As far as I know, node supports it starting in version 15 so I thought it should work. I'll look into using babel and in the mean time just avoid logical assignment.

**Comment by newhouse** [comment]
@ljharb 

Thanks for your comment here!

Is this project still being maintained? I encountered this error and I'm seeing a lot of other issues that are very similar with a common theme: some syntaxes that are supported natively in modern versions of Node are causing parsing errors in browserify / module-deps / detective.

Just roughly speaking it seems like browserify is not keeping up with the syntax of the more recent Node versions, or does not take into account the "target" Node/browser capabilities when parsing things.

Is there any way to make something like this work today without transpilation to an unnecessarily old syntax?

Or is there a way to transpile the code just-in-time for browserify only to figure out the deps?

Thanks!

**Comment by ljharb** [comment]
@newhouse yes, it's still being maintained. I agree that it's not keeping up, but that's not the same thing. Transpilation is pretty universally necessary anyways, so I'd suggest using a babel browserify transform as a workaround, while modern syntax support is pending in all the various deps of browserify.

**Comment by newhouse** [comment]
@ljharb Thanks for the reply.

Yeah I mean in my case I was already transpiling to a target that matched the runtime Node version of my code, but now that I've bumped the runtime I've also bumped the target and this error just started happening.

Since I was already transpiling I just kept an older target for now so it was not a big deal, but:
- it should not be necessary
- eventually something that can't be transpiled will cause a problem that the workaround can't solve
- native optimizations will be lost
- users writing in a syntax thats 100% supported by their runtime will find it more than annoying to have to use babel unnecessarily

Anyways, I appreciate this library, its complexity, your response, and the likely fact that all the things I just said you probably already knew. I just wanted to enumerate them a bit to have them out there.

I'm not even using this library directly - another dependency is - but is there a way to somehow for a User transform the code in just the offending spot in order get the desired results while not actually having the resulting code have been transpiled as well? Probably not but just wondering. And again I'm sorry I'm not actually super familiar with this library and am not directly consuming it.

**Comment by ljharb** [comment]
I suppose you could, yeah - you could probably run babel on a browserify bundle with the only enabled transforms being for the syntaxes you want to target? Seems awkward though.

**Comment by myocytebd** [comment]
The dependent package: `module-deps` seems unmaintained and outdated. It need update on deps: `detective-typescript`.

**Comment by mikixing** [comment]
您的邮件我已收到，会尽快查阅~

**Comment by ljharb** [comment]
It's not, and it's not, and it uses `detective`. I have no idea what `detective-typescript` is.

**Comment by cyberixae** [comment]
Babel doesn't help if the execution never reaches Babel. Try the following for example.
```sh
mkdir tmp && cd tmp && npm init -y
npm install browserify babelify effect
npx --no-install browserify -t babelify node_modules/effect/dist/cjs/GlobalValue.js
```

**Comment by rmaenk** [comment]
### Reason of the issue:
`browserify` uses `insert-module-globals` module, which doesn't pass options to `parse` method of `acorn-node` module. In result , `acorn-node` module uses hardcoded default options:
```js
{
    ecmaVersion: 2020,
    allowHashBang: true,
    allowReturnOutsideFunction: true
}
```
Logical assignment (??=, ||=, &&=) has been introduced in [ECMAScript 2021](https://github.com/daumann/ECMAScript-new-features-list/blob/main/ES2021.MD). So, `ecmaVersion` option is lower than it is required.

### Hotfix:
If it is possible, then change default hardcoded options in `acorn-node` module:
```js
function mapOptions (opts) {
  if (!opts) opts = {}
  return xtend({
    ecmaVersion: 2022,//Hotfix: was 2020, may be set to "latest"
    allowHashBang: true,
    allowReturnOutsideFunction: true
  }, opts)
}
```
Alternatively - make a fork of `acorn-node` module and tag/release. Do the change for `ecmaVersion` option.
In package.json add overrides:
```json
  "overrides": {
    "insert-module-globals": {
      "acorn-node": "github:yourrepository/acorn-node#2.0.1"
    }
}
```
Re-install `insert-module-globals` module to get forked `acorn-node` module.

**Comment by mikixing** [comment]
您的邮件我已收到，会尽快查阅~

**Comment by ljharb** [comment]
Seems like `insert-module-globals` is thus the module that needs a change. Can someone make a PR there?
